docs/sources/whatsnew/whats-new-in-v9-5.md

```diff
@@ -190,3 +190,15 @@ We updated the reporting UI to better fit the new navigation style, adding a hor
 _Experimental in Grafana Open Source and Enterprise._
 
 This feature adds support for using JWT tokens to store rendering keys instead of relying on “remote caching”. It covers most rendering use cases, though some still rely on the remote cache as a store. You can enable this by enabling the feature flag `renderAuthJWT` in the `custom.ini` configuration file.
+
+## Note for plugin developers
+
+One of the major changes in Grafana 10 will be our upgrade to React 18 and use of the new React client rendering API. There are many huge benefits we gain from this: access to new React features like [transitions](https://react.dev/reference/react/useTransition) and concurrent rendering, as well as other general performance and security improvements.
+
+As with any major upgrade, there is a potential for this to impact the way your plugin works. In particular, there could be unintended side effects due to the changes around improving consistency with `useEffect` timings and automatic batching of state updates. These changes have now been delivered to the core grafana repo with [this PR](https://github.com/grafana/grafana/pull/64428).
+
+Recommended actions:
+
+- Review the React 18 [upgrade docs](https://react.dev/blog/2022/03/08/react-18-upgrade-guide)
+- Test your plugins against one of the latest [grafana-dev docker images](https://hub.docker.com/r/grafana/grafana-dev/tags?page=1) (e.g. [this one](https://hub.docker.com/layers/grafana/grafana-dev/10.0.0-111404pre/images/sha256-ac78acf54b44bd2ce7e68b796b1df47030da7f35e53b02bc3eec3f4de05f780f?context=explore))
+- Add a comment to the [forum discussion](link) if your plugin is impacted in any way. Either to socialise the changes needed for your plugin or to reach out and ask for help yourself.
```

Comment:
waiting for approval on these docs internally before adding a forum post. once it's added, i'll update the link here